app/helpers/course_summaries_helper.rb

```diff
@@ -0,0 +1,193 @@
+    if (memberships.count == 0)
+      return nil
+    end
+    get_grouping_for_assignment_for_membership(memberships, assignment)
+  end
+
+  def get_grouping_for_assignment_for_membership(memberships, assignment)
+    memberships.each do |membership|
+      grouping = Grouping.where(id: membership.grouping_id).first
+      if (grouping.assignment_id == assignment.id)
+        return grouping
+      end
+    end
+    nil
+  end
+
+  # Return an object that contains an key for each assignment
+  # and the value is the max mark or null if no marking scheme
+  # is currently defined
+  def get_max_mark_for_assignments
+    max_marks = {}
+    Assignment.all.each do |a|
+      max_marks[a.id] = get_max_mark_for_assignment(a.id)
+    end
+    max_marks.to_json
+  end
+
+  # Get max mark for assignment with id a_id
+  def get_max_mark_for_assignment(a_id)
+    rubric_criterias = RubricCriterion.where(assignment_id: a_id)
+    if (rubric_criterias.count == 0)
+      return nil
+    end
+    max_mark = 0
+    rubric_criterias.each do |rc|
+      max_mark += rc.weight * 4
+    end
+    max_mark
+  end
+
+  def get_mark_for_all_gef_for_student(student, all_grade_entry_forms)
+    marks = {}
+    all_grade_entry_forms.each do |gef|
+      marks[gef.id] = get_mark_for_gef_and_student(gef, student)
+    end
+    marks
+  end
+
+  def get_mark_for_gef_and_student(gef, student)
+    ges = GradeEntryStudent.where(grade_entry_form_id: gef.id,
+                                  user_id: student.id)
+    if ges != []
+      return ges.first.total_grade
+    end
+    nil
+  end
+
+  def get_max_mark_for_grade_entry_forms
+    max_marks = {}
+    GradeEntryForm.all.each do |gef|
+      max_marks[gef.id] = get_max_mark_for_grade_entry_from(gef.id)
+    end
+    max_marks
+  end
+
+  def get_max_mark_for_grade_entry_from(gef_id)
+    total = 0
+    GradeEntryItem.where(grade_entry_form_id: gef_id).each do |gei|
+      total += gei.out_of
+    end
+    total
+  end
+
+  # Return an object that contains a key for each marking
+  # scheme. And the value is an object with keys as assignment ids
+  # and the value the weight for that assignment
+  def get_marking_weights_for_all_marking_schemes
+    result = {}
+    MarkingScheme.all.each do |ms|
+      result[ms.id] = {}
+      MarkingWeight.where(marking_scheme_id: ms.id).each do |mw|
+        result[ms.id][mw.a_id] = mw.weight
+      end
+    end
+    result
+  end
+
+  def get_weighted_total_for_all_marking_schemes_for_student(student)
+    result = {}
+
+    assignment_marks =
+      get_mark_for_all_assignments_for_student(student, Assignment.all)
+    gef_marks = get_mark_for_all_gef_for_student(student, GradeEntryForm.all)
+
+    MarkingScheme.all.each do |ms|
+      result[ms.id] = get_weighted_total_for_marking_scheme_and_student(
+        ms.id,
+        assignment_marks,
+        gef_marks)
+    end
+    result
+  end
+
+  def get_weighted_total_for_marking_scheme_and_student(
```

Comment:
Assignment Branch Condition size for get_weighted_total_for_marking_scheme_and_student is too high. [27.09/15]